internal/ingress/backend/endpoint.go

```diff
@@ -69,15 +69,16 @@ func (resolver *endpointResolver) resolveInstance(ingress *extensions.Ingress, b
 		instanceID, err := resolver.store.GetNodeInstanceID(node)
 		if err != nil {
 			return nil, err
-		} else if healthy, err := resolver.cloud.IsNodeHealthy(instanceID); err != nil {
-			return nil, err
-		} else if !healthy {
-			continue
 		}
-		result = append(result, &elbv2.TargetDescription{
-			Id:   aws.String(instanceID),
-			Port: aws.Int64(int64(nodePort)),
-		})
+		for _, nodeCondition := range node.Status.Conditions {
+			if nodeCondition.Type == corev1.NodeReady && nodeCondition.Status == corev1.ConditionTrue {
```

Comment:
Might be nice (in a future PR, not necessary in this one) to add a `nodeIsReady()` function that hides the above two lines. That would make this function easier to read:

```go
        for _, node := range resolver.store.ListNodes() {
		instanceID, err := resolver.store.GetNodeInstanceID(node)
		if err != nil {
			return nil, err
                }
                if nodeIsReady(node) {
		        result = append(result, &elbv2.TargetDescription{
		                Id:   aws.String(instanceID),
			        Port: aws.Int64(int64(nodePort)),
		        })
                }
        }
```